src/context/data.tsx

```diff
@@ -83,36 +80,52 @@ export const DataProvider: React.FC<React.PropsWithChildren> = ({
 
   // create state for synchronization
   const [synced, setSynced] = useState<boolean>(false);
+
   //TODO: Use real values instread.
-  const [filterValues, setFilterValues] = useState<FilterValues>({
-    radius: { lower: 4, upper: 50 },
-    height: { lower: 3, upper: 43 },
-  });
+  const [filterValues, setCurrentFilterValues] = useState<FilterValues | undefined>(undefined);
 
   // use the offline context
   const { inventory } = useOffline();
 
+  // define the setFilterValues function
+  const setFilterValues = (newValues: FilterValues) => {
+    setCurrentFilterValues({
+      height: { ...newValues.height },
+      radius: { ...newValues.radius}
+    })
+  } 
+
   // copy over inventory data
   useEffect(() => {
     if (inventory) {
+      // copy inventory over
       setAllInventory(inventory);
+
+      // calculate the stats for variables
+      const currentStats = {
+        radiusMax: Math.max(
+          ...inventory.features.map((f) => f.properties.radius)
+        ),
+        radiusMin: Math.min(
+          ...inventory.features.map((f) => f.properties.radius)
+        ),
+        heightMax: Math.max(
+          ...inventory.features.map((f) => f.properties.height)
+        ),
+        heightMin: Math.min(
+          ...inventory.features.map((f) => f.properties.height)
+        ),
+      }
       setInventoryStats({
-        data: {
-          radiusMax: Math.max(
-            ...inventory.features.map((f) => f.properties.radius)
-          ),
-          radiusMin: Math.min(
-            ...inventory.features.map((f) => f.properties.radius)
-          ),
-          heightMax: Math.max(
-            ...inventory.features.map((f) => f.properties.height)
-          ),
-          heightMin: Math.min(
-            ...inventory.features.map((f) => f.properties.height)
-          ),
-        },
+        data: {...currentStats},
       });
 
+      // set the filter to the current min/max as the dataset changed
+      setCurrentFilterValues({
+        radius: {lower: currentStats.radiusMin, upper: currentStats.radiusMax},
+        height: {lower: currentStats.heightMin, upper: currentStats.heightMax}
+      })
+
```

Comment:
slight refactoring to be able to use the newly created min and max values also for the current filter. This means, whenever the underlying dataset is updated, the filter will also be reseted, what makes a lot of sense as filter values might otherwise lie outside of the new value ranges.